src/data/player.py

```diff
@@ -643,6 +528,318 @@ def set_board(self, board_id: int, board_number: int, color: BoardColor):
+                        "<b>1.4.4</b> A maximum of 3/5 of the opponents may come from the applicant's federation."
+                    )
+            else:
+                if num_feds < 2:
+                    res.not_enough_federations = msg
+            res.from_own_federations_count = own_count
+            res.from_host_federations_count = federations_counter.get(
+                Federation(self.event.federation), 0
+            )
+            res.federations_count = num_feds
+
+            # Too many in one federation
+            if federations_counter:
+                top_fed, top_count = federations_counter.most_common(1)[0]
+                max_fed = tn.maximum_of_one_federation(rounds)
+                if top_count > max_fed:
+                    res.too_many_one_federation = (
+                        top_fed,
+                        _(
+                            '<b>1.4.4</b> A maximum of 2/3 of the opponents from one federation.'
+                        ),
+                    )
+
+            # Title holders criterion
+            num_titles = sum(titles_counter.values())
+            if num_titles < tn.minimum_title_holders(rounds):
+                res.not_enough_title_holders = _(
+                    '<b>1.4.5a</b> At least 50% of the opponents shall be title-holders, excluding CM and WCM.'
+                )
+
+            res.num_title_holders = num_titles
+            res.title_counts = titles_counter
+
+            # Required titles criterion
+            req = required_titles.get(tn, Counter())
+            total_req = sum(req.values())
+            if total_req < tn.minimum_required_titles(self.tournament):
+                res.not_enough_required_titles = _(
+                    '<b>1.4.5</b> For this norm, at least {min} opponents must have these title(s): {titles}'
+                ).format(
+                    min=tn.minimum_required_titles(self.tournament),
+                    titles=', '.join(str(title) for title in req.keys()),
+                )
+            res.required_titles = list(req.keys())
+            res.required_titles_met = total_req
+
+            # Score criterion
+            if score < TitleNorm.minimum_score(rounds):
+                res.score_too_low = _(
+                    '<b>1.4.8b</b> The minimum score is 35% for all norms.'
+                )
+
+            res.score = score
+
+        # Rating / performance criteria
+        opponents.sort(key=attrgetter('rating'))
+
+        for tn, res in results.items():
+            rating_list = [
+                PlayerRatingAndType(
+                    opponent.rating
+                    if opponent.rating_type == PlayerRatingType.FIDE
+                    else 1400,
+                    opponent.rating_type,
+                )
+                for opponent in opponents
+            ]
+
+            # Minimum rating floor
+            if rating_list and rating_list[0].value < tn.minimum_rating:
```

Comment:
You can't be sure that an unrated opponent will always have a higher rating than any rated opponent.
You need to also sort on the rating type (always putting FIDE > National and FIDE > Estimated)